domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs

```diff
@@ -106,14 +115,14 @@ public async Task<ActionResult<TransferAgreementProposalResponse>> GetTransferAg
             return NotFound();
         }
 
-        var user = new UserDescriptor(User);
+        accessDescriptor.AssertAuthorizedToAccessOrganization(organizationId);
```

Comment:
I would put this as first line in api endpoint. No need to ask database if you don't have access.

Reply:
I just deleted old code, and renamed the controller. that is all